src/_pytest/python.py

```diff
@@ -1343,6 +1343,12 @@ def _idval(
         # Name of a class, function, module, etc.
         name: str = getattr(val, "__name__")
         return name
+    # TODO: Remove type-ignore after next mypy release.
+    # https://github.com/python/typeshed/commit/076983eec45e739c68551cb6119fd7d85fd4afa9
+    elif isinstance(val, os.PathLike):  # type: ignore[misc]
+        path: Union[str, bytes] = os.fspath(val)
+        return path.decode("utf-8") if isinstance(path, bytes) else path
```

Comment:
The `path.decode("utf-8")` is not viable for this, because at least on unix, a bytes path does not have to be valid UTF-8, so the decode may fail.

Reply:
ah yeah, this isn't the way to go. Thanks again for reviewing, I'm happy to wait and see if there's some consensus from your suggestions on the desired behaviour for escaping here before pushing more changes.

Reply:
I've got a working implementation using your suggestion above with 1aae6c1 (it could probably be tidier, I'll touch it up tomorrow)